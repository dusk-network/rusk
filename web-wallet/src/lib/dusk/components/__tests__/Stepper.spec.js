import { afterAll, afterEach, describe, expect, it, vi } from "vitest";
import { cleanup, render } from "@testing-library/svelte";
import { mdiCheckDecagramOutline } from "@mdi/js";

import { Stepper } from "..";

vi.mock("$lib/dusk/string", async (importOriginal) => {
  /** @type {typeof import("$lib/dusk/string")} */
  const original = await importOriginal();

  return {
    ...original,
    randomUUID: () => "some-generated-id",
  };
});

describe("Stepper", () => {
  const baseProps = {
    activeStep: 2,
    steps: [
      { label: "foo" },
      { label: "bar" },
      { label: "baz" },
      { label: "qux" },
      { iconPath: mdiCheckDecagramOutline, label: "quux" },
    ],
  };
  const baseOptions = {
    props: baseProps,
    target: document.body,
  };

  afterEach(cleanup);

  afterAll(() => {
    vi.doUnmock("$lib/dusk/string");
  });

  it("should render the `Stepper` component accepting an array of `StepperStep` objects as steps", async () => {
    const { container, rerender } = render(Stepper, baseOptions);

    expect(container.firstChild).toMatchSnapshot();

    await rerender({ ...baseProps, activeStep: 3 });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the `Stepper` component accepting a number as the amount of steps", async () => {
    const props = { ...baseProps, steps: 5 };
    const { container, rerender } = render(Stepper, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();

    await rerender({ ...baseProps, activeStep: 3 });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should render the `Stepper` with all step labels being displayed, if the right flag is provided", async () => {
    const props = { ...baseProps, showStepLabelWhenInactive: true, steps: 5 };
    const { container } = render(Stepper, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should pass additional class names and attributes to the rendered element", () => {
    const props = {
      ...baseProps,
      className: "foo bar",
      id: "some-id",
    };
    const { container } = render(Stepper, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should add the proper class name for the desired variant", () => {
    const props = {
      ...baseProps,

      /** @type {StepperVariant} */
      variant: "secondary",
    };
    const { container } = render(Stepper, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should allow to hide the step numbers", () => {
    const props = {
      ...baseProps,
      showStepNumbers: false,
    };
    const { container } = render(Stepper, { ...baseOptions, props });

    expect(container.firstChild).toMatchSnapshot();
  });
});
