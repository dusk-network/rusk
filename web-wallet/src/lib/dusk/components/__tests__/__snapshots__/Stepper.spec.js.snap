// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Stepper > should add the proper class name for the desired variant 1`] = `
<div
  class="dusk-stepper dusk-stepper--variant--secondary"
  style="--columns: 5; --progress-width: 40%;"
>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    foo
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    bar
     
  </span>
  <span
    aria-current="step"
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
  </span>
   
  <span
    class="dusk-stepper__step-label"
    id="step-some-generated-id"
  >
    baz
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    4
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    qux
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    <svg
      class="dusk-icon dusk-icon--size--default"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M23 12L20.6 9.2L20.9 5.5L17.3 4.7L15.4 1.5L12 3L8.6 1.5L6.7 4.7L3.1 5.5L3.4 9.2L1 12L3.4 14.8L3.1 18.5L6.7 19.3L8.6 22.5L12 21L15.4 22.5L17.3 19.3L20.9 18.5L20.6 14.8L23 12M18.7 16.9L16 17.5L14.6 19.9L12 18.8L9.4 19.9L8 17.5L5.3 16.9L5.5 14.1L3.7 12L5.5 9.9L5.3 7.1L8 6.5L9.4 4.1L12 5.2L14.6 4.1L16 6.5L18.7 7.1L18.5 9.9L20.3 12L18.5 14.1L18.7 16.9M16.6 7.6L18 9L10 17L6 13L7.4 11.6L10 14.2L16.6 7.6Z"
      />
    </svg>
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    quux
     
  </span>
  
</div>
`;

exports[`Stepper > should allow to hide the step numbers 1`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary"
  style="--columns: 5; --progress-width: 40%;"
>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    foo
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    bar
     
  </span>
  <span
    aria-current="step"
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    
  </span>
   
  <span
    class="dusk-stepper__step-label"
    id="step-some-generated-id"
  >
    baz
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    qux
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    <svg
      class="dusk-icon dusk-icon--size--default"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M23 12L20.6 9.2L20.9 5.5L17.3 4.7L15.4 1.5L12 3L8.6 1.5L6.7 4.7L3.1 5.5L3.4 9.2L1 12L3.4 14.8L3.1 18.5L6.7 19.3L8.6 22.5L12 21L15.4 22.5L17.3 19.3L20.9 18.5L20.6 14.8L23 12M18.7 16.9L16 17.5L14.6 19.9L12 18.8L9.4 19.9L8 17.5L5.3 16.9L5.5 14.1L3.7 12L5.5 9.9L5.3 7.1L8 6.5L9.4 4.1L12 5.2L14.6 4.1L16 6.5L18.7 7.1L18.5 9.9L20.3 12L18.5 14.1L18.7 16.9M16.6 7.6L18 9L10 17L6 13L7.4 11.6L10 14.2L16.6 7.6Z"
      />
    </svg>
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    quux
     
  </span>
  
</div>
`;

exports[`Stepper > should pass additional class names and attributes to the rendered element 1`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary foo bar"
  id="some-id"
  style="--columns: 5; --progress-width: 40%;"
>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    foo
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    bar
     
  </span>
  <span
    aria-current="step"
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
  </span>
   
  <span
    class="dusk-stepper__step-label"
    id="step-some-generated-id"
  >
    baz
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    4
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    qux
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    <svg
      class="dusk-icon dusk-icon--size--default"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M23 12L20.6 9.2L20.9 5.5L17.3 4.7L15.4 1.5L12 3L8.6 1.5L6.7 4.7L3.1 5.5L3.4 9.2L1 12L3.4 14.8L3.1 18.5L6.7 19.3L8.6 22.5L12 21L15.4 22.5L17.3 19.3L20.9 18.5L20.6 14.8L23 12M18.7 16.9L16 17.5L14.6 19.9L12 18.8L9.4 19.9L8 17.5L5.3 16.9L5.5 14.1L3.7 12L5.5 9.9L5.3 7.1L8 6.5L9.4 4.1L12 5.2L14.6 4.1L16 6.5L18.7 7.1L18.5 9.9L20.3 12L18.5 14.1L18.7 16.9M16.6 7.6L18 9L10 17L6 13L7.4 11.6L10 14.2L16.6 7.6Z"
      />
    </svg>
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    quux
     
  </span>
  
</div>
`;

exports[`Stepper > should render the \`Stepper\` component accepting a number as the amount of steps 1`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary"
  style="--columns: 5; --progress-width: 40%;"
>
  <span
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
     
  </span>
  <span
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
     
  </span>
  <span
    aria-current="step"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
     
  </span>
  <span
    class="dusk-stepper__step"
  >
    4
     
  </span>
  <span
    class="dusk-stepper__step"
  >
    5
     
  </span>
  
</div>
`;

exports[`Stepper > should render the \`Stepper\` component accepting a number as the amount of steps 2`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary"
  style="--columns: 5; --progress-width: 60%;"
>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    foo
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    bar
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    baz
     
  </span>
  <span
    aria-current="step"
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    4
  </span>
   
  <span
    class="dusk-stepper__step-label"
    id="step-some-generated-id"
  >
    qux
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    <svg
      class="dusk-icon dusk-icon--size--default"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M23 12L20.6 9.2L20.9 5.5L17.3 4.7L15.4 1.5L12 3L8.6 1.5L6.7 4.7L3.1 5.5L3.4 9.2L1 12L3.4 14.8L3.1 18.5L6.7 19.3L8.6 22.5L12 21L15.4 22.5L17.3 19.3L20.9 18.5L20.6 14.8L23 12M18.7 16.9L16 17.5L14.6 19.9L12 18.8L9.4 19.9L8 17.5L5.3 16.9L5.5 14.1L3.7 12L5.5 9.9L5.3 7.1L8 6.5L9.4 4.1L12 5.2L14.6 4.1L16 6.5L18.7 7.1L18.5 9.9L20.3 12L18.5 14.1L18.7 16.9M16.6 7.6L18 9L10 17L6 13L7.4 11.6L10 14.2L16.6 7.6Z"
      />
    </svg>
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    quux
     
  </span>
  
</div>
`;

exports[`Stepper > should render the \`Stepper\` component accepting an array of \`StepperStep\` objects as steps 1`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary"
  style="--columns: 5; --progress-width: 40%;"
>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    foo
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    bar
     
  </span>
  <span
    aria-current="step"
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
  </span>
   
  <span
    class="dusk-stepper__step-label"
    id="step-some-generated-id"
  >
    baz
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    4
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    qux
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    <svg
      class="dusk-icon dusk-icon--size--default"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M23 12L20.6 9.2L20.9 5.5L17.3 4.7L15.4 1.5L12 3L8.6 1.5L6.7 4.7L3.1 5.5L3.4 9.2L1 12L3.4 14.8L3.1 18.5L6.7 19.3L8.6 22.5L12 21L15.4 22.5L17.3 19.3L20.9 18.5L20.6 14.8L23 12M18.7 16.9L16 17.5L14.6 19.9L12 18.8L9.4 19.9L8 17.5L5.3 16.9L5.5 14.1L3.7 12L5.5 9.9L5.3 7.1L8 6.5L9.4 4.1L12 5.2L14.6 4.1L16 6.5L18.7 7.1L18.5 9.9L20.3 12L18.5 14.1L18.7 16.9M16.6 7.6L18 9L10 17L6 13L7.4 11.6L10 14.2L16.6 7.6Z"
      />
    </svg>
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    quux
     
  </span>
  
</div>
`;

exports[`Stepper > should render the \`Stepper\` component accepting an array of \`StepperStep\` objects as steps 2`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary"
  style="--columns: 5; --progress-width: 60%;"
>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    foo
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    bar
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    baz
     
  </span>
  <span
    aria-current="step"
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    4
  </span>
   
  <span
    class="dusk-stepper__step-label"
    id="step-some-generated-id"
  >
    qux
     
  </span>
  <span
    aria-labelledby="step-some-generated-id"
    class="dusk-stepper__step"
  >
    <svg
      class="dusk-icon dusk-icon--size--default"
      role="graphics-symbol"
      viewBox="0 0 24 24"
    >
      <path
        d="M23 12L20.6 9.2L20.9 5.5L17.3 4.7L15.4 1.5L12 3L8.6 1.5L6.7 4.7L3.1 5.5L3.4 9.2L1 12L3.4 14.8L3.1 18.5L6.7 19.3L8.6 22.5L12 21L15.4 22.5L17.3 19.3L20.9 18.5L20.6 14.8L23 12M18.7 16.9L16 17.5L14.6 19.9L12 18.8L9.4 19.9L8 17.5L5.3 16.9L5.5 14.1L3.7 12L5.5 9.9L5.3 7.1L8 6.5L9.4 4.1L12 5.2L14.6 4.1L16 6.5L18.7 7.1L18.5 9.9L20.3 12L18.5 14.1L18.7 16.9M16.6 7.6L18 9L10 17L6 13L7.4 11.6L10 14.2L16.6 7.6Z"
      />
    </svg>
    
  </span>
   
  <span
    class="dusk-stepper__step-label dusk-stepper__step-label--invisible"
    id="step-some-generated-id"
  >
    quux
     
  </span>
  
</div>
`;

exports[`Stepper > should render the \`Stepper\` with all step labels being displayed, if the right flag is provided 1`] = `
<div
  class="dusk-stepper dusk-stepper--variant--primary"
  style="--columns: 5; --progress-width: 40%;"
>
  <span
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    1
     
  </span>
  <span
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    2
     
  </span>
  <span
    aria-current="step"
    class="dusk-stepper__step dusk-stepper__step--processed"
  >
    3
     
  </span>
  <span
    class="dusk-stepper__step"
  >
    4
     
  </span>
  <span
    class="dusk-stepper__step"
  >
    5
     
  </span>
  
</div>
`;
